From c2cff65623a99a90e4a08a433107282b38823a1b Mon Sep 17 00:00:00 2001
From: Jasper Mattsson <jasu@njomotys.info>
Date: Sat, 25 Aug 2018 15:35:00 +0300
Subject: [PATCH 3/3] dts: mediatek: mt6797: Bind audio, mjc, and venc clocks
 to SCPSYS

Add newly supported SCPSYS clocks to mt6797.dtsi and devicetree bindings
documentation.

Signed-off-by: Jasper Mattsson <jasu@njomotys.info>
---
 .../devicetree/bindings/soc/mediatek/scpsys.txt          | 2 +-
 arch/arm64/boot/dts/mediatek/mt6797.dtsi                 | 9 ++++++---
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/Documentation/devicetree/bindings/soc/mediatek/scpsys.txt b/Documentation/devicetree/bindings/soc/mediatek/scpsys.txt
index d6fe16f09..2d603481e 100644
--- a/Documentation/devicetree/bindings/soc/mediatek/scpsys.txt
+++ b/Documentation/devicetree/bindings/soc/mediatek/scpsys.txt
@@ -32,7 +32,7 @@ Required properties:
                       enabled before enabling certain power domains.
 	Required clocks for MT2701 or MT7623: "mm", "mfg", "ethif"
 	Required clocks for MT2712: "mm", "mfg", "venc", "jpgdec", "audio", "vdec"
-	Required clocks for MT6797: "mm", "mfg", "vdec"
+	Required clocks for MT6797: "mm", "mfg", "venc", "mjc", "audio", "vdec"
 	Required clocks for MT7622: "hif_sel"
 	Required clocks for MT7622A: "ethif"
 	Required clocks for MT8173: "mm", "mfg", "venc", "venc_lt"
diff --git a/arch/arm64/boot/dts/mediatek/mt6797.dtsi b/arch/arm64/boot/dts/mediatek/mt6797.dtsi
index 4beaa7110..9c05c4fa2 100644
--- a/arch/arm64/boot/dts/mediatek/mt6797.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt6797.dtsi
@@ -133,10 +133,13 @@
 		compatible = "mediatek,mt6797-scpsys";
 		#power-domain-cells = <1>;
 		reg = <0 0x10006000 0 0x1000>;
-		clocks = <&topckgen CLK_TOP_MUX_MFG>,
+		clocks = <&infrasys CLK_INFRA_AUDIO_26M>,
+			 <&topckgen CLK_TOP_MUX_MFG>,
+			 <&topckgen CLK_TOP_MUX_MJC>,
 			 <&topckgen CLK_TOP_MUX_MM>,
-			 <&topckgen CLK_TOP_MUX_VDEC>;
-		clock-names = "mfg", "mm", "vdec";
+			 <&topckgen CLK_TOP_MUX_VDEC>,
+			 <&topckgen CLK_TOP_MUX_VENC>;
+		clock-names = "audio", "mfg", "mjc", "mm", "vdec", "venc";
 		infracfg = <&infrasys>;
 	};
 
-- 
2.18.0

